boutdata/tests/test_collect.py

```diff
@@ -0,0 +1,2773 @@
+        grid_info["MZ"] = grid_info["nz"]
+        grid_info["ixseps1"] = 7
+        grid_info["ixseps2"] = 7
+        grid_info["jyseps1_1"] = -1
+        grid_info["jyseps2_1"] = grid_info["ny"] // 2 - 1
+        grid_info["ny_inner"] = grid_info["ny"] // 2
+        grid_info["jyseps1_2"] = grid_info["ny"] // 2 - 1
+        grid_info["jyseps2_2"] = grid_info["ny"] // 2
+
+        fieldperp_global_yind = 3
+        fieldperp_yproc_ind = 0
+
+        rng = np.random.default_rng(100)
+
+        # core
+        # core includes "ylower" and "yupper" even though there is no actual y-boundary
+        # because collect/squashoutput collect these points
+        dump_params = [
+            {
+                "i": 0,
+                "boundaries": ["xinner", "xouter", "ylower", "yupper"],
+                "fieldperp_global_yind": fieldperp_global_yind,
+            },
+        ]
+        dumps = []
+        for p in dump_params:
+            dumps.append(
+                create_dump_file(
+                    tmpdir=tmp_path,
+                    rng=rng,
+                    grid_info=grid_info,
+                    **p,
+                )
+            )
+
+        expected = concatenate_data(
+            dumps, nxpe=grid_info["NXPE"], fieldperp_yproc_ind=fieldperp_yproc_ind
+        )
+
+        check_collected_data(
+            expected,
+            fieldperp_global_yind=fieldperp_global_yind,
+            doublenull=False,
+            path=tmp_path,
+            squash=squash,
+            collect_kwargs=collect_kwargs,
+        )
+
+    @pytest.mark.parametrize("squash", [False, True])
+    @pytest.mark.parametrize("collect_kwargs", collect_kwargs_list)
+    def test_core(self, tmp_path, squash, collect_kwargs):
+        grid_info = {}
+        grid_info["iteration"] = 6
+        grid_info["MXSUB"] = 3
+        grid_info["MYSUB"] = 4
+        grid_info["MZSUB"] = 5
+        grid_info["MXG"] = 2
+        grid_info["MYG"] = 2
+        grid_info["MZG"] = 0
+        grid_info["NXPE"] = 3
+        grid_info["NYPE"] = 3
+        grid_info["NZPE"] = 1
+        grid_info["nx"] = grid_info["NXPE"] * grid_info["MXSUB"] + 2 * grid_info["MXG"]
+        grid_info["ny"] = grid_info["NYPE"] * grid_info["MYSUB"]
+        grid_info["nz"] = grid_info["NZPE"] * grid_info["MZSUB"]
+        grid_info["MZ"] = grid_info["nz"]
+        grid_info["ixseps1"] = 7
+        grid_info["ixseps2"] = 7
+        grid_info["jyseps1_1"] = -1
+        grid_info["jyseps2_1"] = grid_info["ny"] // 2 - 1
+        grid_info["ny_inner"] = grid_info["ny"] // 2
+        grid_info["jyseps1_2"] = grid_info["ny"] // 2 - 1
+        grid_info["jyseps2_2"] = grid_info["ny"] // 2
+
+        fieldperp_global_yind = 3
+        fieldperp_yproc_ind = 0
+
+        rng = np.random.default_rng(101)
+
+        # core
+        # core includes "ylower" and "yupper" even though there is no actual y-boundary
+        # because collect/squashoutput collect these points
+        dump_params = [
```

Comment:
Is `dump_params` also often duplicated in the tests? Would it make sense to have a class member which is only slightly changed from time to time?

Reply:
I think it makes sense to keep `dump_params` as is. Those parameters are setting out the processor layout manually in a reasonably compact way. I don't want to replicate logic from `collect()` when testing `collect()`, so I think giving them explicitly in each test is the best.

Reply:
Yes, after looking again, it looks like they are varying quite a bit